src/common/mod.rs

```diff
@@ -323,6 +323,43 @@ where
         Poll::Ready(Ok(pos))
     }
 
+    fn poll_write_vectored(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+        bufs: &[IoSlice<'_>],
+    ) -> Poll<io::Result<usize>> {
+        if !bufs.iter().any(|buf| !buf.is_empty()) {
+            return Poll::Ready(Ok(0));
+        }
+
+        loop {
+            let mut would_block = false;
+            let written = self.session.writer().write_vectored(bufs)?;
+
+            while self.session.wants_write() {
+                match self.write_io(cx) {
+                    Poll::Ready(Ok(0)) | Poll::Pending => {
+                        would_block = true;
+                        break;
+                    }
+                    Poll::Ready(Ok(_)) => (),
+                    Poll::Ready(Err(err)) => return Poll::Ready(Err(err)),
+                }
+            }
+
+            return match (written, would_block) {
+                (0, true) => Poll::Pending,
+                (0, false) => continue,
```

Comment:
This looks like a dangerous infinite loop

Reply:
Instead of looping, you could also do:

```rust
cx.waker().wake_by_ref();
return Poll::Pending;
```

Reply:
I thought about it. I think the initial version is fine but needs to be checked for empty input.

but problem with hang is not with poll_write_vectored, but rather with openssl not seeming to output expected text.

Reply:
I went back to the previous version, which also checked for empty input. I'll try seeing if I can fix the test